dataprofiler/profilers/categorical_column_profile.py

```diff
@@ -192,10 +200,24 @@ def categorical_counts(self) -> dict[str, int]:
     @property
     def unique_ratio(self) -> float:
         """Return ratio of unique categories to sample_size."""
+        if self._stop_condition_is_met:
+            return self._unique_ratio
+
         unique_ratio = 1.0
         if self.sample_size:
             unique_ratio = len(self.categories) / self.sample_size
-        return unique_ratio
+        self._unique_ratio = unique_ratio
+        return self._unique_ratio
+
+    @property
+    def unique_count(self) -> float:
+        """Return ratio of unique categories to sample_size."""
+        if self._stop_condition_is_met:
+            return self._unique_count
+
+        self._unique_count = len(self.categories)
```

Comment:
let's only set `self._unique_count` if the stop condition was met. we can handle this below in the _update as described in the other comment

Reply:
I think we need to update the hidden naming of the var as referenced below to declare the difference.
something like `self._stopped_at_unique_count`